ENVParser/EnvFile.cs

```diff
@@ -0,0 +1,580 @@
+        public bool DisableUnknownFlaggedSection { get; set; }
+        public float FieldShadowFarClip { get; set; }
+        public float Field294 { get; set; }
+        public float AmbientShadowBrightness { get; set; }
+        public float Field29C { get; set; }
+        public uint Field2A0 { get; set; }
+        public float FieldShadowNearClip { get; set; }
+        public float FieldShadowBrightness { get; set; }
+        public bool Field2AC { get; set; }
+        public bool Field2AD { get; set; }
+        public bool Field2AE { get; set; }
+        public bool Field2AF { get; set; }
+        public float ShadowColourRed { get; set; }
+        public float ShadowColourGreen { get; set; }
+        public float ShadowColourBlue { get; set; }
+        public float ShadowColourAlpha { get; set; }
+        public bool DisplayColourGrading { get; set; }
+        public float Cyan { get; set; }
+        public float Magenta { get; set; }
+        public float Yellow { get; set; }
+        public float Dodge { get; set; }
+        public float Burn { get; set; }
+        public float LightMapR { get; set; }
+        public float LightMapG { get; set; }
+        public float LightMapB { get; set; }
+        public float LightMapA { get; set; }
+        public bool EnableOutline { get; set; }
+        public float OutlineOpacity { get; set; }
+        public float OutlineWidth { get; set; }
+        public float CharacterOutlineBrightness { get; set; }
+        public float Field2F2 { get; set; }
+        public float Field2F6 { get; set; }
+        public float ReflectionHeight { get; set; }
+        public bool EnablePhysicsSection { get; set; }
+        public float Gravity { get; set; }
+        public bool EnableWind { get; set; }
+        public float WindDirectionX { get; set; }
+        public float WindDirectionY { get; set; }
+        public float WindDirectionZ { get; set; }
+        public float WindStrength { get; set; }
+        public float WindStrengthModifier { get; set; }
+        public float WindCycleTime { get; set; }
+        public float WindCycleDelay { get; set; }
+        public byte ClearColourRed { get; set; }
+        public byte ClearColourGreen { get; set; }
+        public byte ClearColourBlue { get; set; }
+        public byte ClearColourAlpha { get; set; }
+        public uint Field324 { get; set; }
+        public uint Field328 { get; set; }
+        public uint Field32C { get; set; }
+        public uint Field330 { get; set; }
+        public uint Field334 { get; set; }
+        public byte Field338 { get; set; }
+
+        public EnvFile Read(BigEndianBinaryReader reader)
+        {
+            FileMagic = reader.ReadUInt32();
+            GFSVersion = reader.ReadUInt32();
+            FileType = reader.ReadUInt32();
+            Field0C = reader.ReadUInt32();
+            Field10 = reader.ReadByte();
+            EnableFieldModelSection = reader.ReadBoolean();
```

Comment:
You could probably save some space by putting the below FieldModel fields into a class called something like `ModelLighting`, so EnvFile would contain one `ModelLighting` object called `FieldModelLighting` and one called `CharacterModelLighting` (to similarly contain the CharacterModel fields below). It looks like both of these sections have all the same fields, so it would cut down on some repetition.

Reply:
I was thinking for a general refactor, to maybe split it out into the various subheadings altogether (so maybe a class for FieldModel Settings, then CharacterModel Settings), and then within that, yeah use what you've suggested.

Reply:
I like that idea a lot! But from the sounds of that, you want to make it a separate PR. And I think that's reasonable.